notebooks/instructlab-knowledge/instructlab-knowledge.ipynb

```diff
@@ -36,68 +130,130 @@
     "WORKSPACE_ROOT.mkdir(exist_ok=True)\n",
     "\n",
     "WORKSPACE_DIR = WORKSPACE_ROOT / WORKSPACE_NAME\n",
-    "WORKSPACE_DIR.mkdir(exist_ok=True)\n",
+    "WORKSPACE_DIR.mkdir(exist_ok=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "412d5a43-4ec4-43e5-8f08-21aae6c69bfd",
+   "metadata": {},
+   "source": [
+    "To create contributions, define the `name` for the contribution, and the `domain` and `summary`. The `name`, `domain` and `summary` go into a dictionary called `knowledge_contribution` which gets added to a list called `contributions`.\n",
     "\n",
-    "SOURCE_DOCUMENT = None # to process a specific document, set its path here; otherwise, the entire source documents repository will be used\n",
-    "SOURCE_DOCUMENT_DIR = WORKSPACE_DIR / \"source_documents\"\n",
-    "SOURCE_DOCUMENT_DIR.mkdir(parents=True, exist_ok=True)\n",
+    "Once the list of `contributions` is set, a directory with each contribution name is created within the workspace and subdirectories for `source_documents`, `conversion`, `chunking`, `authoring` are created within the contribution name directory."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8b440e34-817c-4588-9a8c-790f74ec5dbb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Populated later on\n",
+    "contributions = []\n",
```

Comment:
I like the idea of packaging these ups as "contributions". Since every `contribution` has the same set of properties, I think it would be clearer to use a class with structured fields in place of a dictionary here. It would significantly lower the margin of error, and make errors a lot easier to resolve.